pkg/imager/profile/output.go

```diff
@@ -97,3 +105,55 @@ const (
 	SDBootEnrollKeysForce                          // force
 	SDBootEnrollKeysOff                            // off
 )
+
+// DiskImageBootloader is a bootloader for the disk image.
+type DiskImageBootloader int
+
+const (
+	// DiskImageBootloaderDualBoot is the dual-boot bootloader
+	// using sd-boot for UEFI and GRUB for BIOS.
+	DiskImageBootloaderDualBoot DiskImageBootloader = iota // dual-boot
+	// DiskImageBootloaderSDBoot is the sd-boot bootloader.
+	DiskImageBootloaderSDBoot // sd-boot
+	// DiskImageBootloaderGrub is the GRUB bootloader.
+	DiskImageBootloaderGrub // grub
+)
+
+// FillDefaults fills default values for the output.
+func (o *Output) FillDefaults(arch, version string, secureboot bool) {
+	if o.Kind == OutKindImage {
+		if o.ImageOptions == nil {
+			o.ImageOptions = &ImageOptions{}
+		}
+
+		// allow user to override bootloader
+		if o.ImageOptions.Bootloader > 0 {
+			return
+		}
+
+		if secureboot {
+			o.ImageOptions.Bootloader = DiskImageBootloaderSDBoot
+
+			return
+		}
+
+		useSDBoot := quirks.New(version).UseSDBootForUEFI()
+
+		// for arm64, we always use sd-boot
+		if arch == "arm64" && useSDBoot {
```

Comment:
I'm almost certain it should be
```suggestion
		if arch == "arm64" || useSDBoot {
```

Reply:
actually this is correct, we only want sd-boot for arm64, for 1.10 amd64 the default is dual-boot